### ISSUE TITLE
Looks like we still have a concurrency issue in our unit tests: concurrent map writes

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#1213]
I think this is likely due to concurrent unit tests running again?

```
=== RUN   TestVolumeCreateGetListAndDelete
time="2016-06-25T15:51:16Z" level=info msg="Refreshing volume cache." 
fatal error: concurrent map writes

goroutine 10 [running]:
runtime.throw(0x8781a0, 0x15)
	/usr/local/go/src/runtime/panic.go:547 +0x90 fp=0xc820032db8 sp=0xc820032da0
runtime.mapassign1(0x719300, 0xc820178570, 0xc820032f78, 0xc820032ee8)
	/usr/local/go/src/runtime/hashmap.go:445 +0xb1 fp=0xc820032e60 sp=0xc820032db8
github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete.func1(0x1)
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:125 +0x3ac fp=0xc820032fa8 sp=0xc820032e60
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc820032fb0 sp=0xc820032fa8
created by github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:132 +0x4db

goroutine 1 [chan receive]:
testing.RunTests(0x8e2720, 0xa27460, 0x6, 0x6, 0xc820074a01)
	/usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc820047ef8, 0xa2ac60)
	/usr/local/go/src/testing/testing.go:515 +0x81
main.main()
	github.com/vmware/vic/lib/portlayer/storage/_test/_testmain.go:118 +0x253

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 8 [semacquire]:
sync.runtime_Semacquire(0xc82017a66c)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*WaitGroup).Wait(0xc82017a660)
	/usr/local/go/src/sync/waitgroup.go:127 +0xb4
github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete(0xc82007a360)
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:134 +0x4fe
testing.tRunner(0xc82007a360, 0xa274a8)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892

goroutine 9 [runnable]:
github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete.func1(0x0)
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:114
created by github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:132 +0x4db

goroutine 11 [runnable]:
github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete.func1(0x2)
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:114
created by github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:132 +0x4db

goroutine 12 [runnable]:
github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete.func1(0x3)
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:114
created by github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:132 +0x4db

goroutine 13 [runnable]:
github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete.func1(0x4)
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:125 +0x3ac
created by github.com/vmware/vic/lib/portlayer/storage.TestVolumeCreateGetListAndDelete
	/drone/src/github.com/vmware/vic/lib/portlayer/storage/volume_cache_test.go:132 +0x4db
exit status 2
FAIL	github.com/vmware/vic/lib/portlayer/storage	0.027s
Makefile:197: recipe for target 'test' failed
```

**Comment by hickeng** [comment]
@fdawg4l can you take a look at this and determine the severity

**Comment by fdawg4l** [comment]
Issue in the unit test itself.  Will fix.  Very minor.  To be fair, the code being exercised isn't even being called yet so it's doubly minor.

**Comment by fdawg4l** [comment]
Closed by #1196
